### DISCUSSION TITLE
Migrating from Valetudo and very old firmware

### DISCUSSION THREAD

**Discussion by betterworld2** [rand256/valetudo#385] [Q&A]
Hi,
I would like to migrate my S5, firmware version 3.3.9, build 1518 (by the way, for some reason the timer on that S5 has stopped a couple of months ago, I hope the migration will fix this) to Valetudo Re.
Can you tell me how I should do this?

Is there still an issue with random resets?

**Comment by pidator** [comment]
You can either build your own firmwarefile by using [this builder](https://github.com/zvldz/vacuum) or use one of the prebuild images available at [release section](https://github.com/rand256/valetudo/releases) and simply flash the new firmware with a tool of your choice (e.g. mirobo tools with the command `mirobo --ip <ROBOT IP> --token <CURRENT TOKEN> update-firmware <firmwarefile.pkg>`).

Then you've updated your robot base firmware and Valetudo RE to the latest version.

Edit:
while building your one firmware using the `--fix-reset` flag solves the random reset problem, in prebuild images this is already included too!

**Reply by betterworld2** [reply]
Thank you!

Is there a difference between these builders?
https://builder.dontvacuum.me/_s5.html
https://github.com/rand256/vacuum
https://github.com/zvldz/vacuum

**Reply by rand256** [reply]
Yes, those are quiet different.

The builder on builder.dontvacuum.me is the origin one, it's focused on security and creates images that don't have preset passwords, so you can only ssh to them with a keyfile you either upload yourself of have auto-generated. Thus it generates personal images per each user - you get a download link in a email.

The builder by vzldz creates images with a preset password, which is up to user to change right after the first login. It also has a lot of options and addons to choose from. I test RE versions with images created by this builder.

And my copy of zvldz's builder is just an old copy. I probably should delete it. Please don't use it.

**Reply by pidator** [reply]
> The builder on builder.dontvacuum.me is the origin one, it's focused on security

Afaik it focuses on another way to bypass the miio-client restrictions in comparison with the zvldz builder, too. I'm currently aware of at least two S5 installations with Valetudo RE where the map didn't get displayed ("no map data").

> And my copy of zvldz's builder is just an old copy. I probably should delete it. 

I think so, yes :)

**Reply by betterworld2** [reply]
> Afaik it focuses on another way to bypass the miio-client restrictions in comparison with the zvldz builder, too. I'm currently aware of at least two S5 installations with Valetudo RE where the map didn't get displayed ("no map data").

you mean installations that used builder.dontvacuumme, so I shouldn't use that?

**Reply by pidator** [reply]
> you mean installations that used builder.dontvacuumme

Yes. After using the zvldz builder everything works fine then (at least with a RE installation).

**Reply by betterworld2** [reply]
great, thanks! Will use this then!

**Reply by rand256** [reply]
> Afaik it focuses on another way to bypass the miio-client restrictions in comparison with the zvldz builder, too. I'm currently aware of at least two S5 installations with Valetudo RE where the map didn't get displayed ("no map data").

Well, in this case I guess dustbuilder may patch miio-client for origin valetudo and do nothing for RE supposing it may workaround that by itself and it can't, thus the results.

**Reply by sanderant** [reply]
Piggy backing on this thread.. Is Roborock Control Center interchangeable with mirobo tools for updating firmware? I know it's not developed any more, but does it use the same process or is mirobo tools safer? 

Thanks for all your hard work!
